lib/pure/parseopt.nim

```diff
@@ -456,7 +456,7 @@ iterator getopt*(p: var OptParser): tuple[kind: CmdLineKind, key,
     yield (p.kind, p.key, p.val)
 
 when (NimMajor, NimMinor, NimPatch) >= (1, 5, 1) or declared(initOptParser):
-  iterator getopt*(cmdline: seq[string] = commandLineParams(),
+  iterator getopt*(cmdline: seq[string] = @[],
```

Comment:
This is a breaking change for non-JS targets ?. 
`getopt()` without args returns `commandLineParams()`.  :thinking:

Reply:
I don't think this is breaking, this `cmdline` argument is only used to pass to `initOptParser`, and [commandLineParams](https://github.com/nim-lang/Nim/blob/version-1-4/lib/pure/os.nim#L2791) is basically the same as what `initOptParser` turns an empty seq into. Might be missing something though?

Reply:
Right now you can call `getopt()` without arguments and it automatically takes the `commandLineParams()` not empty `seq`.

This is a public API, changing an argument that returns the actual value to hardcoded empty value is a breaking change.

Reply:
I'm saying the empty seq is equivalent to `commandLineParams()`. I'm saying there is no difference, `initOptParser` treats empty seq by doing exactly what `commandLineParams()` does and uses that as the command line. The doc comment below this `getopt` definition even specifically says this

Reply:
this special case (and the one in `initOptParser`) is a bad API.
Suddenly you have a weird edge case in code like this:
```nim
let cmdline = commandLineParams().postProcess() # removes some flags for example
for a in getopt(cmdline): handle(a)
```
=> buggy in case `cmdline` happens to be empty after `postProcess`, as it re-introduces `commandLineParams` which differs from `cmdline`

`initOptParser` also has same problem.

one way to fix would be as follows: use a special value, eg `"D20210211T173231"` or `@["D20210211T173231"]` depending on the overload, and expose it as a default:
```nim
let useCmdLine* = "D20210211T173231"
proc initOptParser*(cmdline = useCmdLine, ...)
```

Reply:
=> filed https://github.com/timotheecour/Nim/issues/587